### DISCUSSION TITLE
How to use native android view

### DISCUSSION THREAD

**Discussion by malciin** [dotnet/maui#5065] [Q&A]
Hello. I'm struggling with using native android view. I didn't find any documentation how to achieve that, so I tried initially to follow up approach from Xamarin.Forms, so I added following code to have `MainActivity.Instance` property:

```csharp
    public class MainActivity : MauiAppCompatActivity
    {
        public static MainActivity Instance { get; private set; }

        protected override void OnCreate(Bundle savedInstanceState)
        {
            Instance = this;

            base.OnCreate(savedInstanceState);
            Platform.Init(Instance, savedInstanceState);
        }
    }
```

And I follow up https://docs.microsoft.com/en-us/xamarin/xamarin-forms/platform/native-views/xaml so I created:

```xaml
<?xml version="1.0" encoding="utf-8" ?>
<ContentView xmlns="http://schemas.microsoft.com/dotnet/2021/maui"
             xmlns:x="http://schemas.microsoft.com/winfx/2009/xaml"
             x:Class="BlahBlah"
             xmlns:androidWidget="clr-namespace:Android.Widget;assembly=Mono.Android;targetPlatform=Android">
    <ContentView.Content>
        <StackLayout>
            <androidWidget:TextView Text="Hello World" x:Arguments="{x:Static androidLocal:MainActivity.Instance}" />

<!-- Rest omitted -->
```

But I always get exception. I also tried to do programmatically:

```csharp
InitializeComponent();

var textView = new Android.Widget.TextView(MainActivity.Instance) { Text = "test" };
```

but all I can get from both approaches is the exception:  `System.Reflection.TargetInvocationException: 'Exception has been thrown by the target of an invocation.'`. Anyone could give me a working example? Thanks!

**Comment by SarthakGz** [comment]
You should create a View hander which will be returning Android.Widget.TextView from android implementation
This video might help you https://www.youtube.com/watch?v=bWdilOWAEt4
